openedx/core/djangoapps/profile_images/views.py

```diff
@@ -0,0 +1,96 @@
+"""
+"""
+
+from contextlib import closing
+
+from rest_framework import permissions, status
+from rest_framework.authentication import OAuth2Authentication, SessionAuthentication
+from rest_framework.parsers import MultiPartParser, FormParser
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from ..user_api.accounts.api import set_has_profile_image, get_profile_image_names
+
+from .images import validate_uploaded_image, generate_profile_images, remove_profile_images, ImageFileRejected
+
+
+class ProfileImageUploadView(APIView):
+    """
+    Provides a POST endpoint to generate new profile image files for a given
```

Comment:
If these are to be documented APIs, write Swagger documentation and talk to Mark.